proto/v1/activity_service.proto

```diff
@@ -0,0 +1,213 @@
+  // So using {parent=projects/*} is enough for now.
+  option (google.api.http) = {
+    patch: "/v1/{parent=projects/*/issues/*}/activities"
+    body: "activity"
+  };
+  option (google.api.method_signature) = "parent,activity";
+  }
+
+  rpc SearchActivities(SearchActivitiesRequest) returns (SearchActivitiesResponse) {
+  option (google.api.http) = {
+    post: "/v1/{parent=projects/*}/activities"
+    additional_bindings: {
+      post: "/v1/{parent=workspaces/*}/activities"
+    }
+  };
+  option (google.api.method_signature) = "parent";
+  }
+}
+
+message CreateActivityRequest {
+  // The parent resource name.
+  // Format:
+  // projects/{project}
+  string parent = 1 [(google.api.field_behavior) = REQUIRED];
+
+  // The activity to create.
+  // The type must be ACTION_ISSUE_COMMENT_CREATE for now.
+  Activity activity = 2 [(google.api.field_behavior) = REQUIRED];
+}
+
+message UpdateActivityRequest {
+  // The parent resource name.
+  // Format:
+  // projects/{project}
+  string parent = 1 [(google.api.field_behavior) = REQUIRED];
+
+  // The activity to update.
+  // The type must be ACTION_ISSUE_COMMENT_CREATE for now.
+  Activity activity = 2 [(google.api.field_behavior) = REQUIRED];
+}
+
+message SearchActivitiesRequest {
+  // The parent resource name.
+  // Format:
+  // projects/{project}
+  // workspaces/{workspace}
+  string parent = 1 [(google.api.field_behavior) = REQUIRED];
+
+  // filter is the filter to apply on the search activities request,
+  // follow the [ebnf](https://en.wikipedia.org/wiki/Extended_Backus%E2%80%93Naur_form) syntax.
+  // The field only support in filter:
+  // - creator
+  // - container
+  // - level
+  // - action
+  // For example:
+  // Search the activities of type 'ACTION_ISSUE_COMMENT_CREATE' in issue/123: 'action="ACTION_ISSUE_COMMENT_CREATE", container="issue/123"'
+  string filter = 2;
+
+  // Not used. The maximum number of activities to return.
+  // The service may return fewer than this value.
+  // If unspecified, at most 100 activities will be returned.
+  // The maximum value is 1000; values above 1000 will be coerced to 1000.
+  int32 page_size = 3;
+
+  // Not used. A page token, received from a previous `ListActivities` call.
+  // Provide this to retrieve the subsequent page.
+  string page_token = 4;
+}
+
+message SearchActivitiesResponse {
+  // The list of activities.
+  repeated Activity activities= 1;
+
+  // A token to retrieve next page of activities.
+  // Pass this value in the page_token field in the subsequent call to `ListActivities` method
+  // to retrieve the next page of activities.
+  string next_page_token = 2;
+}
+
+message Activity {
+  // The name of the activity, which is the unqiue identifier.
+  // Generated by the server.
+  // Format:
+  // workspaces/{workspace}/activities/{activity}
+  // projects/{projects}/activities/{activity}
+  string name = 1 [(google.api.field_behavior) = OUTPUT_ONLY];
```

Comment:
Does it need a name (resource URI) for the activity?

Reply:
Removed.